src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+        break;
+      case "second":
+        this.second = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "ampm":
+        if (this.hour === "--") {
+          this.ampm = value as AmPm;
+        } else {
+          const hourAsNumber = parseInt(this.hour);
+          switch (value) {
+            case "AM":
+              if (hourAsNumber >= 12) {
+                this.hour = formatNumberAsTimeString(hourAsNumber - 12);
+              }
+              break;
+            case "PM":
+              if (hourAsNumber < 12) {
+                this.hour = formatNumberAsTimeString(hourAsNumber + 12);
+              }
+              break;
+          }
+          this.ampm = value as AmPm;
+        }
+        break;
+    }
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    if (this.hourDisplayFormat === "12") {
+      this.ampm = this.getAmPm();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const iconScale = this.scale === "s" || this.scale === "m" ? "s" : "m";
+    const includeSeconds = this.step !== 60 ? true : false;
+    return (
+      <Host>
+        <div class="time-picker">
+          <div>
+            <span
+              aria-label="increase hour"
+              class="hour-up shift top-left"
+              onClick={this.incrementHour}
+              onKeyDown={this.incrementHour}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevronup" scale={iconScale} />
+            </span>
+            <span
+              aria-label="Hour"
+              aria-valuemax="23"
+              aria-valuemin="1"
+              aria-valuenow={this.hour !== "--" ? parseInt(this.hour) : undefined}
+              aria-valuetext={this.hour !== "--" ? this.hour : undefined}
+              class="hour input"
+              onBlur={this.hourBlurHandler}
+              onFocus={this.focusHandler}
+              onKeyDown={this.hourKeyDownHandler}
+              ref={(el) => (this.hourEl = el)}
+              role="spinbutton"
+              tabIndex={0}
+            >
+              {this.getDisplayHour()}
+            </span>
+            <span
+              aria-label="decrease hour"
+              class="hour-down shift bottom-left"
+              onClick={this.decrementHour}
+              onKeyDown={this.decrementHour}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevrondown" scale={iconScale} />
+            </span>
+          </div>
+          <span class="colon">:</span>
+          <div>
```

Comment:
Are the empty div's needed? Can we add a class on them if for stying?

Reply:
Those divs are needed for the flex display set on the parent `.time-picker` div, otherwise it looks like this
![image](https://user-images.githubusercontent.com/821864/110867558-cef9d580-827b-11eb-8c4b-7aecefa404b5.png)